src/libostree/ostree-repo-commit.c

```diff
@@ -1650,11 +1650,18 @@ ostree_repo_prepare_transaction (OstreeRepo     *self,
 
   g_mutex_lock (&self->txn_lock);
   self->txn.blocksize = stvfsbuf.f_bsize;
-  guint64 reserved_blocks = min_free_space_calculate_reserved_blocks (self, &stvfsbuf);
+  g_assert (self->txn.blocksize != 0);
+  self->reserved_blocks = min_free_space_calculate_reserved_blocks (self, &stvfsbuf);
+  if (self->reserved_blocks > G_MAXUINT64 / self->txn.blocksize)
+    {
+      g_mutex_unlock (&self->txn_lock);
+      return glnx_throw (error, "min-free-space value(in bytes) was too high");
```

Comment:
It would be good to include the maximum permitted value in this error message. For example:
```
return glnx_throw ("min-free-space value is greater than the maximum allowed value of %" G_GUINT64_FORMAT " bytes", G_MAXUINT64 / self->txn.blocksize);
```